pkg/inframanager/p4/service.go

```
@@ -633,38 +645,54 @@ func DeleteServiceRules(ctx context.Context, p4RtC *client.Client,
 
 	switch service.Proto {
 	case "TCP":
+		log.Debugf("Deleting from table TxBalanceTcpTable, service ip: %s, service port: %d, group id: %d",
+			service.ClusterIp, uint16(service.Port), groupID)
+		if err = TxBalanceTcpTable(ctx, p4RtC, service.ClusterIp, uint16(service.Port), groupID, Delete); err != nil {
+			return err
+		}
+		log.Debugf("Deleting from table AsSl3TcpTable, member ids: %v, mod blob ptrs: %v, group id: %d",
+			memberID, modblobPtrDNAT, groupID)
 		if err = AsSl3TcpTable(ctx, p4RtC, memberID, modblobPtrDNAT, groupID, Delete); err != nil {
 			return nil
```

Shouldn't this be return err instead of return nil. 
Same comment for all below calls.